src/common.jl

```diff
@@ -230,169 +271,40 @@ function validate_linear_solver(linear_solver::String)
-
-        push!(A_I, ix + iix - 1)
-        push!(A_J, to_b)
-        push!(A_V, -1)
-    end
-    return
-end
-
 """
-Evaluates the Incidence matrix A given the branches and node of a System.
-
-# Arguments
-- `branches`:
-        vector containing the branches of the considered system (should be AC branches).
-- `buses::Vector{PSY.ACBus}`:
-        vector containing the buses of the considered system.
-
-NOTE:
-- the matrix features all the columns, including the ones related to the
-  reference buses (each column is related to a system's bus).
+Validates that the user bus input is consistent with the ybus axes and the prior reductions.
+Is used to check `irreducible_buses` for `Radial` and `DegreeTwo` reductions and `study_buses` for `WardReduction`.
 """
-function calculate_A_matrix(
-    branches::Vector{<:PSY.ACTransmission},
-    buses::Vector{PSY.ACBus},
-    network_reduction_data::NetworkReductionData,
-)
-    ref_bus_positions = find_slack_positions(buses)
-    bus_lookup = make_ax_ref(buses)
-    reverse_bus_search_map = get_reverse_bus_search_map(network_reduction_data)
-    A_I = Int[]
-    A_J = Int[]
-    A_V = Int8[]
-    reverse_bus_search_map = get_reverse_bus_search_map(network_reduction_data)
-    # build incidence matrix A (lines x buses)
-    for (ix, b) in enumerate(branches)
-        # we offload the logic to separate functions because we need to treat Transformer3W differently
-        _add_branch_to_A_matrix!(b, ix, bus_lookup, A_I, A_J, A_V, reverse_bus_search_map)
-    end
-
-    return SparseArrays.sparse(A_I, A_J, A_V), ref_bus_positions
-end
-
-function _add_branch_to_BA_matrix!(
-    b::PSY.ACTransmission,
-    ix::Int,
-    bus_lookup::Dict{Int, Int},
-    BA_I::Vector{Int},
-    BA_J::Vector{Int},
-    BA_V::Vector{Float64},
-    reverse_bus_search_map::Dict{Int, Int},
-)
-    b_val = PSY.get_series_susceptance(b)
-    fr_b, to_b = get_bus_indices(b, bus_lookup, reverse_bus_search_map)
-    push!(BA_I, fr_b)
-    push!(BA_J, ix)
-    push!(BA_V, b_val)
-
-    push!(BA_I, to_b)
-    push!(BA_J, ix)
-    push!(BA_V, -b_val)
-    return
-end
-
-function _add_branch_to_BA_matrix!(
-    b::PSY.ThreeWindingTransformer,
-    ix::Int,
-    bus_lookup::Dict{Int, Int},
-    BA_I::Vector{Int},
-    BA_J::Vector{Int},
-    BA_V::Vector{Float64},
-    reverse_bus_search_map::Dict{Int, Int},
-)
-    b_vals = PSY.get_series_susceptance(b)
-    for (iix, (fr_b, to_b)) in
-        enumerate(get_bus_indices(b, bus_lookup, reverse_bus_search_map))
-        b_val = b_vals[iix]
-        push!(BA_I, fr_b)
-        push!(BA_J, ix + iix - 1)
-        push!(BA_V, b_val)
-
-        push!(BA_I, to_b)
-        push!(BA_J, ix + iix - 1)
-        push!(BA_V, -b_val)
+function validate_buses(ybus::Ybus, buses::Vector{Int})
+    reverse_bus_search_map = ybus.network_reduction_data.reverse_bus_search_map
+    for bus_no in buses
+        reduced_bus_no = get(reverse_bus_search_map, bus_no, bus_no)
+        if reduced_bus_no ∉ ybus.axes[1]
```

Comment:
For clarity can you make a getter in the Ybus to grab this axes. We have this pattern all over but it isn't immediately obvious what we are doing.